src/services/extension-registry-provider.js

```diff
@@ -112,7 +148,13 @@
                                           if(memo.length >= limit) {
                                             return memo;
                                           }
-                                          memo.push(next);
+                                          push(memo, next);
+                                          // - heaviest falls.
```

Comment:
Confusing comment since I think `weight: 5` is heavier than `weight: 4`, but `weight: 5` should come first?